src/Features/LanguageServer/Protocol/ExternalAccess/SimplifyTypeNamesHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Simplify;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(SimplifyTypeNamesHandler)), Shared]
+[Method(SimplifyTypeNamesMethodName)]
+internal class SimplifyTypeNamesHandler : ILspServiceDocumentRequestHandler<SimplifyTypeNamesParams, string[]?>
```

Comment:
Just in case, we should probably have a strongly typed return POCO, even if for now it just wraps a `string[]`, just so we can avoid `SumType` as long as possible.